core/main_test.ts

```diff
@@ -1260,6 +1260,453 @@ actions:
+          testConfig.namePrefix ? `${testConfig.namePrefix}_A` : "A",
+          testConfig.namePrefix ? `${testConfig.namePrefix}_C` : "C",
+          testConfig.namePrefix ? `${testConfig.namePrefix}_schema_C_assertions_rowConditions` : "schema_C_assertions_rowConditions",
+        ]);
+      })
+
+      test("Assertions added through includeDependentAssertions and explicitly listed in dependencies are deduplicated.", () => {
+        fs.writeFileSync(path.join(projectDir, "definitions/B.sqlx"),
+          `config {
+            type: "table",
+            dependencies: ["A_assert"]
+        }
+        select * from \${ref({name: "A", includeDependentAssertions: true})}
+        select 1 as btest
+        `);
+
+        const result = runMainInVm(coreExecutionRequestFromPath(projectDir));
+
+        expect(result.compile.compiledGraph.graphErrors.compilationErrors).deep.equals([]);
+        expect(asPlainObject(result.compile.compiledGraph.tables[1].dependencyTargets.length)).equals(3);
+        expect(asPlainObject(result.compile.compiledGraph.tables[1].dependencyTargets.flatMap(dependencyTarget => dependencyTarget.name))).deep.equals([
+          testConfig.namePrefix ? `${testConfig.namePrefix}_A_assert` : "A_assert",
+          testConfig.namePrefix ? `${testConfig.namePrefix}_A` : "A",
+          testConfig.namePrefix ? `${testConfig.namePrefix}_schema_A_assertions_rowConditions` : "schema_A_assertions_rowConditions"
+        ]);
+      })
+
+      test("When includeDependentAssertions property in config and ref are set differently for the same dependency, compilation error is thrown.", () => {
+        fs.writeFileSync(path.join(projectDir, "definitions/B.sqlx"),
+          `config {
+            type: "table",
+            dependencies: [{name: "A", includeDependentAssertions: false}, {name: "C", includeDependentAssertions: true}]
+        }
+        select * from \${ref({name: "A", includeDependentAssertions: true})}
+        select * from \${ref({name: "C", includeDependentAssertions: false})}
+        select 1 as btest
+        `);
+        fs.writeFileSync(path.join(projectDir, "definitions/C.sqlx"),
+          `config {
+            type: "table",
+             assertions: {
+               rowConditions: ["test > 1"]
+            }
+          }
+          SELECT 1 as test
+        }
+        `);
+
+        const result = runMainInVm(coreExecutionRequestFromPath(projectDir));
+        expect(result.compile.compiledGraph.graphErrors.compilationErrors.length).deep.equals(2);
+        expect(result.compile.compiledGraph.graphErrors.compilationErrors[0].message).deep.equals(`Conflicting "includeDependentAssertions" properties are not allowed. Dependency A has different values set for this property.`);
+      })
+
+      suite("Action configs", ({ beforeEach }) => {
+        let projectDir: any;
```

Comment:
You can technically have the same forEach for this suite and the one above, which would simplify the action configs below, by not having to include A and its assertion.

Reply:
But I want to create the assertions using action yaml to make sure it is getting mapped correctly this way as well.